src/Runner.Worker/JobExtension.cs

```diff
@@ -74,6 +74,11 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                         {
                             // print out HostName for self-hosted runner
                             context.Output($"Runner name: '{setting.AgentName}'");
+                            VariableValue runnerGroupName = "";
+                            if (message.Variables.TryGetValue("RunnerGroupName", out runnerGroupName))
```

Comment:
```
if (message.Variables.TryGetValue("RunnerGroupName", out VariableValue runnerGroupName))
```

Reply:
`if (message.Variables.TryGetValue("RunnerGroupName", out var runnerGroupName))`

Reply:
we normally namespace these kinds of variables, like `system.runnergroupname`